### ISSUE TITLE
[Proxmox] Cloud-Init & Automation Support

### ISSUE THREAD

**Issue by SprayerGaming** [WemXPro/tracker#358]
### Describe the solution you'd like

Cloud-Init & Virtual Machine cloning

### Describe alternatives you've considered

An ability to support cloud-init feature of Proxmox, users could set their own username, password, DNS, hostname etc... ,  WemX panel could display the IP adress of the actual VM including username & password would be dope tho. 
Automation: ability to clone VM templates instead of installing from an ISO image.

### Additional context

None.

**Comment by EWANZO101** [comment]
hey so about 2 of them things is nothing to do with wemx as cloud-init  would be somthing u make up in your pve node ssh atm secend of all if u have the know how u could add display ip also dns is in proxmox on every vm  as can be seen here 
![Capture](https://github.com/WemXPro/tracker/assets/107235474/17099a3c-662c-4a6a-9bd4-24a31f4905ef)

**Comment by EWANZO101** [comment]
@SprayerGaming this is the only part i am unsure about #users could set their own username, password,# maybe  u can use this example ``` #cloud-config

users:
  - name: defaultuser
    passwd: defaultpassword
    sudo: ALL=(ALL) NOPASSWD:ALL

runcmd:
  - echo "Please change your default password upon first login."
  - su - defaultuser -c "read -p 'Enter new username: ' newuser && sudo useradd -m $newuser && sudo passwd $newuser"
  - su - defaultuser -c "read -s -p 'Enter new password: ' newpass && echo $newpass | sudo passwd --stdin $USER"

**Comment by SprayerGaming** [comment]
Ah okay, these cloud configs would be good. 
And what about making a feature to clone an existing VM instead of installing froman ISO ? So WemX administrator users can set their Proxmox service to be able to clone a VM template by entering the VMID or just setting the package to be installed from an ISO.

**Comment by EWANZO101** [comment]
> Ah okay, these cloud configs would be good. And what about making a feature to clone an existing VM instead of installing froman ISO ? So WemX administrator users can set their Proxmox service to be able to clone a VM template by entering the VMID or just setting the package to be installed from an ISO.

so part of that is already a thing in a dev v of proxmox as its being tested so i think that is going to be a thing if u look at this screen shot 
![Screenshot_2023-12-16_175816](https://github.com/WemXPro/tracker/assets/107235474/eee0ed02-1161-4157-9229-50552e8b2686)
is this what u would be looking at having ? @SprayerGaming

**Comment by SprayerGaming** [comment]
Exactly, and if the user can select their OS template, then the VMID template (that, would be cloned) should also change, so it could determine the OS of the VM whatever somehow. 
Or, if the ISO could be selected, then Proxmox should also set the exact ISO which was selected by the user :)

**Comment by EWANZO101** [comment]
TBH i think the os selelect has not been added back yet and also ISO most ISO LIST THE NAME OF THE OS ON THE ISO
